virtcontainers/kata_agent.go

```diff
@@ -702,8 +703,13 @@ func (k *kataAgent) replaceOCIMountsForStorages(spec *specs.Spec, volumeStorages
 
 			// Create a temporary location to mount the Storage. Mounting to the correct location
 			// will be handled by the OCI mount structure.
+			var path string
 			filename := fmt.Sprintf("%s-%s", uuid.Generate().String(), filepath.Base(m.Destination))
-			path := filepath.Join(kataGuestSharedDir, filename)
+			if v.Driver == kataBlkDevType || v.Driver == kataSCSIDevType {
+				path = filepath.Join(kataGuestStorageDir, filename)
+			} else {
+				path = filepath.Join(kataGuestSharedDir, filename)
```

Comment:
Here I think there is no need no set another kataGuestStorageDir, why not keep using kataGuestSharedDir?

Reply:
I thought using `kataGuestSharedDir`, but it is a 9pfs shared directory, if reuse this dir we can see the mount dir on the host, and I hope that our mount dir is not visible on the host.

Reply:
Yes, the host will see only an empty directory, is there any dangerous for this? If not,  it's better to keep it as simple as possible.